components/gif_picker/components/SearchGrid/SearchGrid.tsx

```diff
@@ -34,27 +36,162 @@ const mapDispatchToProps = ({
+            width: number;
+            height: number;
+        };
+        largeGif: {
+            url: string;
+            size: number;
+            width: number;
+            height: number;
+        };
+        max1mbGif: {
+            url: string;
+            size: number;
+            width: number;
+            height: number;
+        };
+        max2mbGif: {
+            url: string;
+            size: number;
+            width: number;
+            height: number;
+        };
+        max5mbGif: {
+            url: string;
+            size: number;
+            width: number;
+            height: number;
+        };
+        mobile: {
+            url: string;
+            size: number;
+            width: number;
+            height: number;
+        };
+        mobilePoster: {
+            url: string;
+            size: number;
+            width: number;
+            height: number;
+        };
+        mp4: {
+            url: string;
+            size: number;
+            width: number;
+            height: number;
+        };
+        webm: {
+            url: string;
+            size: number;
+            width: number;
+            height: number;
+        };
+        webp: {
+            url: string;
+            size: number;
+            width: number;
+            height: number;
+        };
+    };
+    createDate: number;
+    description: string;
+    dislikes: number;
+    domainWhitelist: string[];
+    extraLemmas: string;
+    frameRate: number;
+    gatekeeper: number;
+    geoWhitelist: string[];
+    gfyId: string;
+    gfyName: string;
+    gfyNumber: string; // Ex: '227423790'
+    gif100px: string;
+    gifUrl: string;
+    hasAudio: boolean;
+    hasTransparency: boolean;
+    height: number;
+    languageCategories: string[ ];
+    languageText: string;
+    likes: number;
+    max1mbGif: string;
+    max2mbGif: string;
+    max5mbGif: string;
+    miniPosterUrl: string;
+    miniUrl: string;
+    mobilePosterUrl: string;
+    mobileUrl: string;
+    mp4Size: number;
+    mp4Url: string;
+    nsfw: number;
+    numFrames: number;
+    posterUrl: string;
+    published: number;
+    rating: string;
+    sitename: string;
+    source: number;
+    tags: string[ ];
+    thumb100PosterUrl: string;
+    title: string;
+    url: string;
+    userData: {
+        followers: number;
+        following: number;
+        name: string;
+        profileImageUrl: string;
+        subscription: number;
+        username: string;
+        verified: boolean;
+        views: number;
+    };
+    userDisplayName: string;
+    username: string;
+    userProfileImageUrl: string;
+    views: number;
+    webmSize: number;
+    webmUrl: string;
+    webpUrl: string;
+    width: number;
+}
+
+export class SearchGrid extends PureComponent<Props, State> {
+    columnsHeights: number[];
+    container: HTMLElement | null;
+    containerHeight: number | undefined;
+    numberOfColumns: number | undefined;
```

Comment:
we tend to use the question mark for these cases
```suggestion
    containerHeight?: number;
    numberOfColumns?: number;
```